src/vtbackend/Functions.h

```diff
@@ -618,10 +623,82 @@ inline auto const& functions() noexcept
     return funcs;
 }
 
+class SupportedSequences
+{
+  public:
+    SupportedSequences()
+    {
+        for (auto [index, seq]: crispy::indexed(functions()))
+        {
+            _supportedSequences[index] = seq;
+        }
+        crispy::sort(
+            _supportedSequences,
+            [](FunctionDefinition const& a, FunctionDefinition const& b) constexpr { return compare(a, b); });
+        _lastIndex = _supportedSequences.size();
+    }
+
+    [[nodiscard]] gsl::span<FunctionDefinition const> getAllSequences() const noexcept
+    {
+        return gsl::span<FunctionDefinition const>{_supportedSequences.cbegin(), _supportedSequences.cend()};
+    }
+
+    [[nodiscard]] gsl::span<FunctionDefinition const> getSupportedSequencesAtCurrentVTLevel() const noexcept
+    {
+        return gsl::span<FunctionDefinition const>{_supportedSequences.cbegin(),
+                                                   _supportedSequences.cbegin() + _lastIndex};
+    }
+
+    void updateVTLevel(VTType vt) noexcept
```

Comment:
```suggestion
    void reset(VTType vt) noexcept
```
We're resetting the set of available VT sequences based on the given VT level. Since this class is all about that, maybe simply call it `reset` then?